src/setup-docker/templates/fabric-docker/commands-generated.sh

```diff
@@ -5,91 +5,40 @@ function installChaincodes() {
+        <% } -%>
+      else
+        echo "Warning! Skipping chaincode '<%= chaincode.name %>' installation. Chaincode directory is empty."
+        echo "Looked in dir: '$CHAINCODES_BASE_DIR/<%= chaincode.directory %>'"
+      fi
     <% }) %>
   <% } -%>
 }
 
 function upgradeChaincode() {
-  chaincodeName="$1"
-  version="$2"
-
-  if [ -z "$chaincodeName" ]; then
-    echo "Error: chaincode name is not provided"
-    exit 1
-  fi
+  local chaincodeName="$1"
+  if [ -z "$chaincodeName" ]; then echo "Error: chaincode name is not provided"; exit 1; fi
 
-  if [ -z "$version" ]; then
-    echo "Error: chaincode version is not provided"
-    exit 1
-  fi
+  local version="$2"
+  if [ -z "$version" ]; then echo "Error: chaincode version is not provided"; exit 1; fi
 
   <% chaincodes.forEach(function(chaincode) { -%>
     if [ "$chaincodeName" = "<%= chaincode.name %>" ]; then
-      chaincodeBuild <% -%>
-        "<%= chaincode.name %>" <% -%>
-        "<%= chaincode.lang %>" <% -%>
-        "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>"
-      <% chaincode.channel.orgs.forEach(function (org) { -%>
-        <% org.peers.forEach(function (peer) { %>
-          printHeadline "Installing '<%= chaincode.name %>' on <%= chaincode.channel.name %>/<%= org.name %>/<%= peer.name %>" "U1F60E"
-          chaincodeInstall <% -%>
-            "cli.<%= org.domain %>" <% -%>
-            "<%= peer.fullAddress %>" <% -%>
-            "<%= chaincode.channel.name %>" <% -%>
-            "<%= chaincode.name %>" <% -%>
-            "$version" <% -%>
-            "<%= chaincode.lang %>" <% -%>
-            "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" <% -%>
-            "<%= rootOrg.ordererHead.fullAddress %>" <% -%>
-            "<%= !networkSettings.tls ? '' : 'crypto/orderer-tlscacerts/tlsca.' + rootOrg.domain+ '-cert.pem' %>"
-        <% }) -%>
-      <% }) -%>
-      printItalics "Upgrading as '<%= chaincode.instantiatingOrg.name %>'. '<%= chaincode.name %>' on channel '<%= chaincode.channel.name %>'" "U1F618"
-      chaincodeUpgrade <% -%>
-        "cli.<%= chaincode.instantiatingOrg.domain %>" <% -%>
-        "<%=  chaincode.instantiatingOrg.headPeer.fullAddress %>" <% -%>
-        "<%= chaincode.channel.name %>" "<%= chaincode.name %>" <% -%>
-        "$version" <% -%>
-        "<%= chaincode.lang %>" <% -%>
-        "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" <% -%>
-        "<%= rootOrg.ordererHead.fullAddress %>" <% -%>
-        '<%- chaincode.init %>' <% -%>
-        "<%- chaincode.endorsement %>" <% -%>
-        "<%= !networkSettings.tls ? '' : 'crypto/orderer-tlscacerts/tlsca.' + rootOrg.domain+ '-cert.pem' %>" <% -%>
-        "<%= chaincode.privateDataConfigFile || '' %>"
+      if [ -n "$(ls "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>")" ]; then
+        <% if (capabilities.isV2) { -%>
+          <%- include('commands-generated/chaincode-install-v2.sh.ejs', { chaincode, rootOrg, networkSettings }); %>
```

Comment:
hm... don't get it. Why if we're using HLF 2.x we include `chaincode-install-v2.sh.ejs` and for older ones it's `chaincode-upgrade-v1.sh.ejs` ? It's a bug or different naming convention? We want to install it here. not to upgrade it.